### ISSUE TITLE
instructor results view: show a 'view by section' warning for bigger courses

### ISSUE THREAD

**Issue by damithc** [TEAMMATES/repo#3024]
For bigger courses (e.g. 200+), the results page can be loaded with a warning status message that 'by sections' view is recommended. The message should show only if the 'all' option is selected.

**Comment by manvijain** [comment]
Hi,
I would like to try to try to solve this issue. Can it be assigned to me?
I went through the code to get an idea of what needs to be done.Am I right in thinking that the number 200+ indicates the sum of length of RespondingStudentList and RespondingInstructorList in the FeedbackSessionAttribute.java Class?

**Comment by LowWeiLin** [comment]
There is a constant where we switch to use AJAX when the number of responses is too high. Maybe we can use that as a trigger for the warning? Or do we want the warning even if ajax is not used?@damithc

**Comment by manvijain** [comment]
Is the constant you are referring to 500? The one which decides the value of the boolean variable shouldCollapsed based on the size of responses and invokes the toggleCollapse() function in InstructorFeedbackResults.js?@LowWeiLin

**Comment by LowWeiLin** [comment]
I think you mean the jsp files? I think that only collapses the panels, if there is too many.

I think the value that determines if ajax is used is DEFAULT_QUERY_RANGE, found in InstructorFeedbackResultsPageAction.java.

@imouto1994 is that right?

**Comment by damithc** [comment]
@imouto1994 is the best person to advise on this issue. :-)

**Comment by manvijain** [comment]
Yes I was referring to the jsp file for feedbackResult's view in which the decision of collapsing the panels is taken.

**Comment by imouto1994** [comment]
For every JSP file in the results page, there will a boolean value `showAll` to determine whether the list of responses has been loaded completely or we should use AJAX. We could use that to check whether we need to show the warning :)

**Comment by manvijain** [comment]
Found it! That's great! Thank you! :)

**Comment by manvijain** [comment]
I introduced a warning statement when "All" is selected. This is how it looks. Is it okay?
![warning](https://cloud.githubusercontent.com/assets/10461781/6456427/1a6633fc-c189-11e4-9bbb-d6e9c55b634f.png)

**Comment by damithc** [comment]
Warning text: "This session seem to have more responses than we can show in one page. Please view one section at at time"
Perhaps use a regular status message, but in 'warning' formatting?

**Comment by manvijain** [comment]
I made the required changes. Is it looking good now?
![warning2](https://cloud.githubusercontent.com/assets/10461781/6456880/f3e1fcf8-c191-11e4-9515-010795d5f0fd.png)

**Comment by damithc** [comment]
I meant a status message like this
![image](https://cloud.githubusercontent.com/assets/1673303/6458098/716dc53a-c1ba-11e4-8c58-a607e8d8100a.png)

**Comment by manvijain** [comment]
Sorry for the misinterpretation. I changed the formatting. Is it in accordance with what is required?
![warning3](https://cloud.githubusercontent.com/assets/10461781/6458726/d124bae6-c1ac-11e4-8b97-31a6ea847914.png)

**Comment by damithc** [comment]
@LowWeiLin can you mentor this issue?

**Comment by LowWeiLin** [comment]
Can change the text to be:
"This session seem to have a large number of responses. It is recommended to view the results one section at at time"
Since it is not the case that we <i>cannot</i> show the responses?

Also, try to use a warning(yellow) box for the message, since it is not an error, but a warning/recommendation.

**Comment by manvijain** [comment]
I changed the text as well as the statusMessage color. Is it okay now?
![warning4](https://cloud.githubusercontent.com/assets/10461781/6459385/1fd7d298-c1b4-11e4-9091-f93db69b5fb3.png)

**Comment by manvijain** [comment]
I ran the test suite and am getting one failed test in for InstructorFeedbackResultsPageUITest. How can I resolve this?
I am attaching a screenshot of the comparision.
![testcasefail](https://cloud.githubusercontent.com/assets/10461781/6461931/dfad380e-c1ca-11e4-8efc-31019745a3fa.png)

**Comment by LowWeiLin** [comment]
You will need to find the failing line in the test, see which html file is it checking against, and change it to have the new expected behaviour.

**Comment by manvijain** [comment]
I introduced the changed code snippet to all feedbackResult html files using ajax and got rid of the comparison failure.
However there still remains an error due to TimeoutException in testFeedbackResponseCommentActions function in the InstructorFeedbackResultsPageUiTest.java but I don't think it is due to the changes made in this issue. How can I proceed to resolve this failure? Is it relevant to this issue?
![timeoutexception](https://cloud.githubusercontent.com/assets/10461781/6478552/81be38ce-c25b-11e4-80a1-c93be4da8f2f.png)
